packages/mcp-auth/src/utils/bearer-www-authenticate-header.ts

```diff
@@ -0,0 +1,41 @@
+import { condString } from '@silverhand/essentials';
+
+/**
+ * A simple implementation for generating WWW-Authenticate response headers
+ * specifically for Bearer authentication scheme, based on RFC 6750.
+ */
+export class BearerWWWAuthenticateHeader {
+  private readonly authParams = new Map<string, string>();
+
+  constructor(params?: Record<string, string>) {
+    if (params) {
+      this.authParams = new Map(Object.entries(params));
+    }
+  }
+
+  /**
+   * Sets an authentication parameter.
+   * If the value is undefined or empty string, the parameter will be ignored.
+   *
+   * @param param The parameter name
+   * @param value The parameter value, if undefined the parameter will not be set
+   */
+  setParameter(param: string, value?: string) {
+    if (value) {
+      this.authParams.set(param, value);
+    }
+    return this;
+  }
+
+  toString() {
+    const params = Array.from(this.authParams.entries())
+      .map(([key, value]) => `${key}="${value}"`)
+      .join(', ');
+
+    return condString(params && `Bearer ${params}`);
```

Comment:
always true?

<img width="100" alt="image" src="https://github.com/user-attachments/assets/43b24a2d-ad55-42d6-a393-2a3fa197f211" />

Reply:
No, should be `Boolean([].join(', ')) === false`